.github/workflows/build.yml

```diff
@@ -1,24 +1,43 @@
-name: build
-on: push
+name: Build
+on:
+  push:
+    branches:
+      - main
 
 jobs:
   build:
-    runs-on: ubuntu-20.04
-    container:
-      image: openjdk:8-jdk
+    strategy:
+      matrix:
+        java: [
+          1.8, # Minimum
+          11,  # LTS
+          16,  # Latest
```

Comment:
Do we really need to test on all these versions? Minimum should be good enough, or even minimum and latest, if there's no forward compatibility for some reason.

Reply:
iirc most people will use lts for custom installs ie multimc

Reply:
What I meant is: Is it necessary to test compilation on all three versions? Java *shouldn't* be breaking old things with new releases (Unless we're using unsupported APIs like `Unsafe`).

Reply:
it's probably not needed but it's a good sanity check so we know it works on multiple versions.